File: BackEnd/eMenka/eMenka.Domain/Classes/FuelCard.cs
using System;
using System.Collections.Generic;

namespace eMenka.Domain.Classes
{
    public class FuelCard
    {
        public int Id { get; set; }
        public string Number { get; set; }
        public Company Company { get; set; }
        public int? CompanyId { get; set; }
        public Vehicle Vehicle { get; set; }
        public int? VehicleId { get; set; }
        public Driver Driver { get; set; }
        public int? DriverId { get; set; }
        public Record Record { get; set; }
        public int? RecordId { get; set; }
        public bool IsBlocked { get; set; }
        public DateTime? BlockingDate { get; set; }
        public string BlockingReason { get; set; }
        public string PinCode { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public IEnumerable<Refill> Refills { get; set; }

    }
}

File: eFenkaFrontend/src/app/models/fuel-card/fuel-card.ts
import {Record} from '../record/record';
import {Driver} from '../driver/driver';
import {Vehicle} from '../vehicle/vehicle';
import {Company} from '../company/company';
import {FuelType} from '../fuel-type/fuel-type';

export class FuelCard {
  id: number;
  fuelCardNumber: string;
  vehicle: Vehicle;
  driver: Driver;
  record: Record;
  company: Company;
  fuelType: FuelType;
  startDate: Date;
  endDate: Date;
  isActive: boolean;
  blockingDate: Date;
  blockingReason: string;
  isBlocked: boolean;
  pinCode: string;

  constructor(id: number, fuelCardNumber: string, vehicle: Vehicle, driver: Driver,
              record: Record, company: Company, startDate: Date, endDate: Date, isActive: boolean) {
    this.id = id;
    this.fuelCardNumber = fuelCardNumber;
    this.vehicle = vehicle;
    this.driver = driver;
    this.record = record;
    this.company = company;
    this.startDate = startDate;
    this.endDate = endDate;
    this.isActive = isActive;
  }
}
